js/layout-javascript/agenda-code.js

```diff
@@ -1229,21 +1232,17 @@ DynamicList.prototype.renderLoopHTML = function () {
         _this.INCREMENTAL_RENDERING_BATCH_SIZE * (renderLoopIndex + 1)
       );
 
+      // Break render cycle if there is no more data
+      if (!nextBatch.length && renderLoopIndex > 0) {
```

Comment:
Why is the `renderLoopIndex > 0` part still necessary? I imagine it isn't needed any more because the UI state update is done outside fo the loop now at lines 1224-1225.

Reply:
We still need it to render "no results found" which is getting rendered in line 1244.